### ISSUE TITLE
vgg16 network  with arcface loss

### ISSUE THREAD

**Issue by jetsmith** [deepinsight/insightface#271]
hi, have you tried vgg16 with arcface loss? what's your accuracy on lfw? i train the vgg16 with arcface loss training only on the vggface2 dataset,  the loss can't converge and it stop at about 6. I checked the code and without arcface loss it works, but there are some problems while adding arcface loss.i am so confused about it. Adam and momentum got the same result. Please help me .Many thanks.

**Comment by jetsmith** [comment]
@zhaowwenzhong  yea, i have tried it using vgg16 without fc layers(have about 4600c classes) and the optimizer is momentum. it converge slowly and it's very strange that the training batch accuracy is only 0.02% while its loss is 4. what's more, the accuracy does not increase as the loss decreases. So i tested the small dataset on the resnet50 + arcface loss without pretrained softmax( 10 classes and each contains 100 samples) and it works. While i change to vgg16, it did not work. Do i need to add BN layer or change the activation func ? any advice?

**Comment by nttstar** [comment]
How did you define the VGG network? Can you provide the symbol file?

**Comment by jetsmith** [comment]
of course, it is based on the models/slim
`def vgg16_face0(inputs,
           num_classes=None,
           is_training=True,
           dropout_keep_prob=0.5,
           spatial_squeeze=True,
           scope='vgg_16',
           global_pool=True):
with tf.variable_scope(scope, 'vgg_16', [inputs]) as sc:
    end_points_collection = sc.original_name_scope + '_end_points'
    # Collect outputs for conv2d, fully_connected and max_pool2d.
    with slim.arg_scope([slim.conv2d, slim.fully_connected, slim.max_pool2d],
                        outputs_collections=end_points_collection):
      net = slim.repeat(inputs, 2, slim.conv2d, 64, [3, 3], scope='conv1')
      net = slim.max_pool2d(net, [2, 2], scope='pool1')
      net = slim.repeat(net, 2, slim.conv2d, 128, [3, 3], scope='conv2')
      net = slim.max_pool2d(net, [2, 2], scope='pool2')
      net = slim.repeat(net, 3, slim.conv2d, 256, [3, 3], scope='conv3')
      net = slim.max_pool2d(net, [2, 2], scope='pool3')
      net = slim.repeat(net, 3, slim.conv2d, 512, [3, 3], scope='conv4')
      net = slim.max_pool2d(net, [2, 2], scope='pool4')
      net = slim.repeat(net, 3, slim.conv2d, 512, [3, 3], scope='conv5')
      net = slim.max_pool2d(net, [2, 2], scope='pool5')

      # # Use conv2d instead of fully_connected layers.
      # net = slim.conv2d(net, 4096, [7, 7], padding=fc_conv_padding, scope='fc6')
      # net = slim.dropout(net, dropout_keep_prob, is_training=is_training,
      #                    scope='dropout6')
      # net = slim.conv2d(net, 4096, [1, 1], scope='fc7')

      # Convert end_points_collection into a end_point dict.
      end_points = slim.utils.convert_collection_to_dict(end_points_collection)
      if global_pool:
        net = tf.reduce_mean(net, [1, 2], keep_dims=True, name='global_pool')
        end_points['global_pool'] = net
      if num_classes:
        # net = slim.dropout(net, dropout_keep_prob, is_training=is_training,
        #                    scope='dropout7')
        net = slim.conv2d(net, num_classes, [1, 1],
                          activation_fn=None,
                          normalizer_fn=None,
                          scope='fc1')
        if spatial_squeeze:
          net = tf.squeeze(net, [1, 2], name='fc1/squeezed')
        end_points[sc.name + '/fc1'] = net
      return net, end_points`

**Comment by nttstar** [comment]
it's not our mxnet implementation.

**Comment by jetsmith** [comment]
@nttstar i also tried the tensorflow version you mentioned [](https://github.com/auroua/InsightFace_TF)
use nets/vgg16.py with arcface loss, there are still some problems. please give me some sugguestions for me, i have struggled in it for a long time. thanks!

**Comment by nttstar** [comment]
Sorry I don't know much about that TF implementation.

**Comment by jetsmith** [comment]
@nttstar did you tried vgg16+arcface loss using mxnet, how is the performance?

**Comment by nttstar** [comment]
No
